components/jdbc-connector/src/main/java/org/wso2/carbon/security/userstore/jdbc/util/ConnectorContextThreadLocal.java

```diff
@@ -0,0 +1,27 @@
+package org.wso2.carbon.security.userstore.jdbc.util;
+
+
+import org.wso2.carbon.security.userstore.jdbc.connector.virtual.VirtualStoreConnector;
+
+
+/**
+ * ConnectorContextThreadLocal
+ *
+ */
+public class ConnectorContextThreadLocal {
+
+    private static ThreadLocal<VirtualStoreConnector> virtualStoreConnectorThreadLocal = new InheritableThreadLocal<>();
```

Comment:
We should not use TheadLocal in C5, and design the methods to accept proper formal parameters